File: languages/php/Bloock/Entity/Integrity/Network.php
<?php

namespace Bloock\Entity\Integrity;

/**
 * Represents a network.
 */
class Network
{
    const ETHEREUM_MAINNET = "ETHEREUM_MAINNET";
    const ETHEREUM_SEPOLIA = "ETHEREUM_SEPOLIA";
    const GNOSIS_CHAIN = "GNOSIS_CHAIN";
    const POLYGON_CHAIN = "POLYGON_CHAIN";

    public static function fromProto(\Bloock\Network $network): string
    {
        switch ($network) {
            case \Bloock\Network::ETHEREUM_SEPOLIA:
                return Network::ETHEREUM_SEPOLIA;
            case \Bloock\Network::GNOSIS_CHAIN:
                return Network::GNOSIS_CHAIN;
            case \Bloock\Network::POLYGON_CHAIN:
                return Network::POLYGON_CHAIN;
            default:
                return Network::ETHEREUM_MAINNET;
        }
    }

    public static function toProto(string $network): int
    {
        switch ($network) {
            case Network::ETHEREUM_SEPOLIA:
                return \Bloock\Network::ETHEREUM_SEPOLIA;
            case Network::GNOSIS_CHAIN:
                return \Bloock\Network::GNOSIS_CHAIN;
            case Network::POLYGON_CHAIN:
                return \Bloock\Network::POLYGON_CHAIN;
            default:
                return \Bloock\Network::ETHEREUM_MAINNET;
        }
    }
}

File: languages/python/bloock/entity/integrity/network.py
from __future__ import annotations
from warnings import warn

from enum import Enum

from bloock._bridge.proto.bloock_config_pb2 import Network as NetworkProto


class Network(Enum):
    """
    Represents a network.
    """
    ETHEREUM_MAINNET = 0
    ETHEREUM_SEPOLIA = 1
    GNOSIS_CHAIN = 2
    POLYGON_CHAIN = 3

    def __int__(self):
        return self.value

    @staticmethod
    def to_proto(network: Network) -> NetworkProto.ValueType:
        if network == Network.ETHEREUM_MAINNET:
            return NetworkProto.ETHEREUM_MAINNET
        elif network == Network.ETHEREUM_SEPOLIA:
            return NetworkProto.ETHEREUM_SEPOLIA
        elif network == Network.GNOSIS_CHAIN:
            return NetworkProto.GNOSIS_CHAIN
        elif network == Network.POLYGON_CHAIN:
            return NetworkProto.POLYGON_CHAIN
